### PR TITLE
fix non-fetchable hashes in openssl dgst -list

### PR DESCRIPTION
**Pull request by GauriSpears**
`openssl dgst -engine gost -list` doesn't show hashes added by gost-engine because they are not supported by EVP_MD_fetch and EVP_get_digestbyname should be used instead. Another example of this issue has been fixed in https://github.com/openssl/openssl/pull/12689
By the way, `openssl enc -engine gost -list` works correctly.

CLA: trivial

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
I'm okay with trivial.

File: apps/dgst.c [hunk 1]
```diff
@@ -489,7 +489,11 @@ static void show_digests(const OBJ_NAME *name, void *arg)
     /* Filter out message digests that we cannot use */
     md = EVP_MD_fetch(app_get0_libctx(), name->name, app_get0_propq());
     if (md == NULL)
-        return;
+    {
```
Comment (t8m) [inline apps/dgst.c:492]: Style nit: This needs to be at the previous line

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]
OK with CLA: trivial.

**Comment by t8m** [comment]
@paulidale please reconfirm

**Comment by paulidale** [comment]
Confirmed.  I don't feel that reconfirmation is necessary for a trivial reformatting like this.

**Comment by t8m** [comment]
Squashed and merged to master, 3.1, and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**